internal/state/ticket.go

```diff
@@ -0,0 +1,63 @@
+package state
+
+import (
+	"errors"
+
+	"github.com/eigerco/strawberry/internal/block"
+	"github.com/eigerco/strawberry/internal/common"
+	"github.com/eigerco/strawberry/internal/crypto"
+)
+
+// Creates a ticket proof as per equation 6.29 in the graypaper (v0.6.4)
+// The ticket extrinic is a sequence of ticket proofs.
+func CreateTicketProof(state *State, privateKey crypto.BandersnatchPrivateKey, attempt uint8) (block.TicketProof, error) {
+	if attempt > common.MaxTicketAttempts {
+		return block.TicketProof{}, errors.New("attempts exceeded")
+	}
+
+	ringProver, err := state.ValidatorState.SafroleState.RingProver(privateKey)
+	defer ringProver.Free()
+	if err != nil {
+		return block.TicketProof{}, err
```

Comment:
wrapping errors

Reply:
You'd prefer wrapping the errors in both cases?

Reply:
yes

Reply:
Even if the error coming in is pretty clear?

Reply:
in go errors don't have a stack trace, this is why wrapping them helps with debugging to know exactly in which place the error happened even if the error is pretty clear

Reply:
my understanding is you generally wrap when you want to add additional context, or if the error is unclear. in this case it will be "unable to create RingVrfProver", do you still think wrapping is worth it in that case?

Reply:
i don't think wrapping here makes sense, but I have added your suggestions for a simpler return :)